homeassistant/components/broadlink/updater.py

```diff
@@ -35,6 +41,68 @@ def get_update_manager(device):
     return update_managers[device.api.type](device)
 
 
+class BroadlinkScout:
+    """Manages device discovery."""
+
+    def __init__(self, hass):
+        """Initialize the scout."""
+        self.hass = hass
+        self.timeout = 5
+        self.coordinator = DataUpdateCoordinator(
+            hass,
+            _LOGGER,
+            name="discovery",
+            update_method=self.async_discover,
+            update_interval=timedelta(minutes=15),
+            request_refresh_debouncer=debounce.Debouncer(
+                hass, _LOGGER, cooldown=self.timeout, immediate=True
+            ),
+        )
+        self.coordinator.async_add_listener(self.update_listener)
+
+    @callback
+    def update_listener(self):
+        """Create config flows for the devices discovered."""
+        devices = self.coordinator.data
+        if not (self.coordinator.last_update_success and devices):
+            return
+
+        entries = self.hass.config_entries.async_entries(DOMAIN)
+        hosts_and_macs = {
+            (get_ip_or_none(entry.data[CONF_HOST]), entry.data[CONF_MAC])
+            for entry in entries
+        }
+        for device in devices:
+            host, mac_addr = device.host[0], device.mac.hex()
+            if (host, mac_addr) in hosts_and_macs:
+                continue
+
+            self.hass.async_create_task(
+                self.hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": config_entries.SOURCE_INTEGRATION_DISCOVERY},
+                    data={
+                        CONF_HOST: device.host[0],
+                        CONF_MAC: device.mac.hex(),
+                        CONF_TYPE: device.devtype,
+                        CONF_NAME: device.name,
+                        ATTR_LOCKED: device.is_locked,
```

Comment:
We shouldn't use `ATTR_` constants for config data.

Reply:
Ok.